dlt/common/storages/file_storage.py

```diff
@@ -251,3 +254,14 @@ def rmtree_del_ro(action: AnyFun, name: str, exc: Any) -> Any:
                 os.rmdir(name)
             else:
                 os.remove(name)
+
+    @staticmethod
+    def open_zipsafe(path: str, mode: str = "r", encoding: Optional[str] = None, **kwargs: Any) -> IO[Any]:
+        """Opens a file using gzip.open if it is a gzip file, otherwise uses open."""
+        try:
+            f = gzip.open(path, mode, encoding=encoding, **kwargs)
```

Comment:
ohhh the encoding will probably not work here! see our CI and common tests. also mode may contain "b" or "t". this needs to be unit tested